server/plugin.go

```diff
@@ -53,12 +54,22 @@ func (p *MatterpollPlugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r
 	}
 }
 
-func (p *MatterpollPlugin) ConvertUserToDisplayName(userID string) (string, *model.AppError) {
+func (p *MatterpollPlugin) ConvertUserIDToDisplayName(userID string) (string, *model.AppError) {
 	user, err := p.API.GetUser(userID)
 	if err != nil {
 		return "", err
 	}
-	// NOTE: We should better fetch the server config and us this instead of model.SHOW_FULLNAME
-	nameFormat := model.SHOW_FULLNAME
-	return user.GetDisplayName(nameFormat), nil
+	displayName := user.GetDisplayName(*p.ServerConfig.TeamSettings.TeammateNameDisplay)
+	if displayName == user.Username {
```

Comment:
When I tested in my local, I may have noticed useful spec of `TeammateNameDisplay` settings.
If server returns `@USERNAME`, client converts `@USERNAME` to the display_name according to `TeammateNameDisplay` settings in AccountSettings. Users can see display names in format users want to see.

So we have better to use `model.SHOW_USERNAME` and prepend `@` always. 

How do you think about this?

Reply:
Oh, thank you for the hint. This kind of kills the purpose of this PR and make it do mostly refactoring. I updated my code and the description of this PR.

Reply:
Thank you for a lot of work and sorry for changing the purpose.
Updated code is LGTM, thanks !